packages/lodestar/src/tasks/index.ts

```diff
@@ -70,8 +74,27 @@ export class TasksService implements IService {
     await this.interopSubnetsTask.stop();
   };
 
-  private handleFinalizedCheckpointChores = async (finalized: BlockSummary, pruned: BlockSummary[]): Promise<void> => {
-    await new ArchiveBlocksTask(this.config, {db: this.db, logger: this.logger}, finalized, pruned).run();
-    await new ArchiveStatesTask(this.config, {db: this.db, logger: this.logger}, finalized, pruned).run();
+  private onFinalizedCheckpoint = async (finalized: Checkpoint): Promise<void> => {
+    await Promise.all([
+      new ArchiveStatesTask(this.config, {db: this.db, logger: this.logger}, finalized).run(),
+      new ArchiveBlocksTask(
```

Comment:
I think we want to make sure ArchiveBlock finish before starting ArchiveTask due to https://github.com/ChainSafe/lodestar/issues/1381